sifter/grammar/state.py

```diff
@@ -1,25 +1,32 @@
-import sifter.grammar
+from typing import (
+    Text,
+    Dict,
+    Optional
+)
+from sifter.grammar.actions import Actions
 
 __all__ = ('EvaluationState',)
 
+
 class EvaluationState(object):
 
-    def __init__(self):
-        self.actions = sifter.grammar.Actions(implicit_keep=True)
-        self.required_extensions = {}
+    def __init__(self) -> None:
+        self.actions = Actions(implicit_keep=True)
+        self.required_extensions: Dict[Text, bool] = {}
+        self.last_if: Optional[bool] = None
```

Comment:
"self.last_if" from EvaluationState is used in CommandIfBase, but was not a member of EvaluationState. So I have added it to the state. By using properties from objects, it is a goot practice to add it before first use.

I set self.last_if to None, because It might not be set, when no if is used in the rules. Perhaps we can add another/better initial value.